pkg/alertmanager/operator.go

```diff
@@ -1086,6 +1089,7 @@ func (c *Operator) selectAlertmanagerConfigs(ctx context.Context, am *monitoring
 				"namespace", am.Namespace,
 				"alertmanager", am.Name,
 			)
+			c.eventRecorder.Eventf(amc, v1.EventTypeWarning, "InvalidConfiguration", "AlertmanagerConfig %q/%q was rejected due to invalid configuration: %v", amc.GetNamespace(), amc.GetName(), err)
```

Comment:
I'm wondering if we need to repeat the namespace of the object since it's already implied via involvedObject? 

```suggestion
			c.eventRecorder.Eventf(amc, v1.EventTypeWarning, "InvalidConfiguration", "AlertmanagerConfig %s was rejected due to invalid configuration: %v", amc.GetName(), err)
```

Reply:
Good point, when using kubectl get events I don't see the namespace attached to other events from common resources, e.g. pod/deployment.

<details><summary>kubectl get events -n monitoring</summary>

```
@ArthurSens ➜ /workspaces/prometheus-operator (continue-pr-6030) $ kubectl get events -n monitoring
LAST SEEN   TYPE      REASON                 OBJECT                                     MESSAGE
27m         Normal    Killing                pod/prometheus-operator-6db7d8bcfd-mxrqg   Stopping container prometheus-operator
27m         Normal    Killing                pod/prometheus-operator-6db7d8bcfd-mxrqg   Stopping container kube-rbac-proxy
24m         Normal    Scheduled              pod/prometheus-operator-78d96f-7mckr       Successfully assigned monitoring/prometheus-operator-78d96f-7mckr to kind-control-plane
24m         Normal    Pulled                 pod/prometheus-operator-78d96f-7mckr       Container image "quay.io/prometheus-operator/prometheus-operator:f8cdf4889" already present on machine
24m         Normal    Created                pod/prometheus-operator-78d96f-7mckr       Created container prometheus-operator
24m         Normal    Started                pod/prometheus-operator-78d96f-7mckr       Started container prometheus-operator
24m         Normal    Pulled                 pod/prometheus-operator-78d96f-7mckr       Container image "quay.io/brancz/kube-rbac-proxy:v0.15.0" already present on machine
24m         Normal    Created                pod/prometheus-operator-78d96f-7mckr       Created container kube-rbac-proxy
24m         Normal    Started                pod/prometheus-operator-78d96f-7mckr       Started container kube-rbac-proxy
20m         Normal    Killing                pod/prometheus-operator-78d96f-7mckr       Stopping container prometheus-operator
20m         Normal    Killing                pod/prometheus-operator-78d96f-7mckr       Stopping container kube-rbac-proxy
27m         Normal    Scheduled              pod/prometheus-operator-78d96f-h2vm4       Successfully assigned monitoring/prometheus-operator-78d96f-h2vm4 to kind-control-plane
27m         Normal    Pulled                 pod/prometheus-operator-78d96f-h2vm4       Container image "quay.io/prometheus-operator/prometheus-operator:f8cdf4889" already present on machine
27m         Normal    Created                pod/prometheus-operator-78d96f-h2vm4       Created container prometheus-operator
27m         Normal    Started                pod/prometheus-operator-78d96f-h2vm4       Started container prometheus-operator
27m         Normal    Pulled                 pod/prometheus-operator-78d96f-h2vm4       Container image "quay.io/brancz/kube-rbac-proxy:v0.15.0" already present on machine
27m         Normal    Created                pod/prometheus-operator-78d96f-h2vm4       Created container kube-rbac-proxy
27m         Normal    Started                pod/prometheus-operator-78d96f-h2vm4       Started container kube-rbac-proxy
24m         Normal    Killing                pod/prometheus-operator-78d96f-h2vm4       Stopping container prometheus-operator
24m         Normal    Killing                pod/prometheus-operator-78d96f-h2vm4       Stopping container kube-rbac-proxy
20m         Normal    Scheduled              pod/prometheus-operator-78d96f-nqp9j       Successfully assigned monitoring/prometheus-operator-78d96f-nqp9j to kind-control-plane
20m         Normal    Pulled                 pod/prometheus-operator-78d96f-nqp9j       Container image "quay.io/prometheus-operator/prometheus-operator:f8cdf4889" already present on machine
20m         Normal    Created                pod/prometheus-operator-78d96f-nqp9j       Created container prometheus-operator
20m         Normal    Started                pod/prometheus-operator-78d96f-nqp9j       Started container prometheus-operator
20m         Normal    Pulled                 pod/prometheus-operator-78d96f-nqp9j       Container image "quay.io/brancz/kube-rbac-proxy:v0.15.0" already present on machine
20m         Normal    Created                pod/prometheus-operator-78d96f-nqp9j       Created container kube-rbac-proxy
20m         Normal    Started                pod/prometheus-operator-78d96f-nqp9j       Started container kube-rbac-proxy
19m         Normal    Killing                pod/prometheus-operator-78d96f-nqp9j       Stopping container prometheus-operator
19m         Normal    Killing                pod/prometheus-operator-78d96f-nqp9j       Stopping container kube-rbac-proxy
27m         Normal    SuccessfulCreate       replicaset/prometheus-operator-78d96f      Created pod: prometheus-operator-78d96f-h2vm4
24m         Normal    SuccessfulCreate       replicaset/prometheus-operator-78d96f      Created pod: prometheus-operator-78d96f-7mckr
20m         Normal    SuccessfulCreate       replicaset/prometheus-operator-78d96f      Created pod: prometheus-operator-78d96f-nqp9j
19m         Normal    SuccessfulDelete       replicaset/prometheus-operator-78d96f      Deleted pod: prometheus-operator-78d96f-nqp9j
20m         Normal    Scheduled              pod/prometheus-operator-b7545ddf6-r9ntp    Successfully assigned monitoring/prometheus-operator-b7545ddf6-r9ntp to kind-control-plane
20m         Normal    Pulled                 pod/prometheus-operator-b7545ddf6-r9ntp    Container image "quay.io/prometheus-operator/prometheus-operator:f8cdf4889" already present on machine
20m         Normal    Created                pod/prometheus-operator-b7545ddf6-r9ntp    Created container prometheus-operator
20m         Normal    Started                pod/prometheus-operator-b7545ddf6-r9ntp    Started container prometheus-operator
20m         Normal    Pulled                 pod/prometheus-operator-b7545ddf6-r9ntp    Container image "quay.io/brancz/kube-rbac-proxy:v0.15.0" already present on machine
20m         Normal    Created                pod/prometheus-operator-b7545ddf6-r9ntp    Created container kube-rbac-proxy
20m         Normal    Started                pod/prometheus-operator-b7545ddf6-r9ntp    Started container kube-rbac-proxy
20m         Normal    SuccessfulCreate       replicaset/prometheus-operator-b7545ddf6   Created pod: prometheus-operator-b7545ddf6-r9ntp
27m         Normal    ScalingReplicaSet      deployment/prometheus-operator             Scaled up replica set prometheus-operator-78d96f to 1
24m         Normal    ScalingReplicaSet      deployment/prometheus-operator             Scaled up replica set prometheus-operator-78d96f to 1
20m         Normal    ScalingReplicaSet      deployment/prometheus-operator             Scaled up replica set prometheus-operator-78d96f to 1
20m         Warning   InvalidConfiguration   servicemonitor/prometheus-operator         ServiceMonitor "monitoring"/"prometheus-operator" was rejected due to invalid configuration: failed to get bearer token: failed to get token from secret: unable to get secret "nonExistent": secrets "nonExistent" not found
20m         Normal    ScalingReplicaSet      deployment/prometheus-operator             Scaled up replica set prometheus-operator-b7545ddf6 to 1
20m         Warning   InvalidConfiguration   servicemonitor/prometheus-operator         ServiceMonitor "monitoring"/"prometheus-operator" was rejected due to invalid configuration: failed to get bearer token: failed to get token from secret: unable to get secret "nonExistent": secrets "nonExistent" not found
19m         Normal    ScalingReplicaSet      deployment/prometheus-operator             Scaled down replica set prometheus-operator-78d96f to 0 from 1
```

</details>